ikarus/finiteelements/feresulttypes.hh

```diff
@@ -0,0 +1,146 @@
+    auto operator()(const Eigen::MatrixBase<Derived>& mat) const {
+      return mat.derived().reshaped().eval();
+    }
+  };
+
+  struct MatricizeWithVoigt
+  {
+    template <typename Derived>
+    auto operator()(const Eigen::MatrixBase<Derived>& vec) const {
+      return fromVoigt(vec.derived(), false);
+    }
+  };
+
+  struct MatricizeGeneric
+  {
+    template <typename Derived>
+    auto operator()(const Eigen::MatrixBase<Derived>& vec) const {
+      auto rc = static_cast<int>(std::floor(std::sqrt(vec.derived().size())));
+      return vec.derived().reshaped(rc, rc).eval();
+    }
+  };
+
+} // namespace Util
+
+namespace ResultType {
+#define REGISTER_RESULTTYPE_IMPL(structName, rowsExpr, ColsExpr, VectorizeStruct, MatricizeStruct) \
+  template <typename ScalarType, int gridDim, int worldDim>                                        \
+  struct structName                                                                                \
+  {                                                                                                \
+    friend auto toString(structName) { return #structName; }                                       \
+                                                                                                   \
+    using type       = Eigen::Matrix<ScalarType, rowsExpr, ColsExpr>;                              \
+    using Vectorizer = VectorizeStruct;                                                            \
+    using Matricizer = MatricizeStruct;                                                            \
+  }
+
+#define REGISTER_SYMMETRIC_RESULTTYPE(structName, rowsExpr, ColsExpr)                        \
+  REGISTER_RESULTTYPE_IMPL(structName, rowsExpr, ColsExpr, Ikarus::Util::VectorizeWithVoigt, \
+                           Ikarus::Util::MatricizeWithVoigt)
+
+#define REGISTER_RESULTTYPE(structName, rowsExpr, ColsExpr)                                \
+  REGISTER_RESULTTYPE_IMPL(structName, rowsExpr, ColsExpr, Ikarus::Util::VectorizeGeneric, \
+                           Ikarus::Util::MatricizeGeneric)
+
+  ///=============================
+  /// Registered ResultTypes
+
+  REGISTER_SYMMETRIC_RESULTTYPE(linearStress, worldDim, worldDim);
+  REGISTER_SYMMETRIC_RESULTTYPE(PK2Stress, worldDim, worldDim);
+  REGISTER_SYMMETRIC_RESULTTYPE(cauchyStress, worldDim, worldDim);
+
+  REGISTER_RESULTTYPE(director, worldDim, 1);
+  REGISTER_RESULTTYPE(magnetization, worldDim, 1);
+  REGISTER_RESULTTYPE(gradientNormOfMagnetization, 1, 1);
+  REGISTER_RESULTTYPE(vectorPotential, worldDim, 1);
+  REGISTER_RESULTTYPE(divergenceOfVectorPotential, 1, 1);
+
+  REGISTER_RESULTTYPE(BField, 1, 1);
+  REGISTER_RESULTTYPE(HField, 1, 1);
+
+  REGISTER_RESULTTYPE(customType, Eigen::Dynamic, Eigen::Dynamic);
+} // namespace ResultType
+
+template <template <typename, int, int> class RT1, template <typename, int, int> class RT2>
+constexpr bool isSameResultType = std::is_same_v<RT1<double, 1, 1>, RT2<double, 1, 1>>;
```

Comment:
If you are wondering how to do this more general: https://stackoverflow.com/a/52322078/12484740.

typename... but has to be `typename, int, int`,  there is no really general way to do this